students/guillaume/quarter_2_project/leankitbackend/leankitbackend/leankitapi.py

```diff
@@ -0,0 +1,259 @@
+            auth = HTTPBasicAuth(args['username'], args['password'])
+
+        if token_id is not None:
+            token_lst = [token_id]
+        else:
+            self.list_tokens(**args)
+            token_lst = [token['id'] for token in self.active_tokens]
+
+        for token in token_lst:
+            url = "https://{}{}/{}".format(args['company'], web_auth, token)
+            response = delete(url, auth=auth)
+            self.url = response.url
+            self.list_tokens(**args)
+            self.status = response.status_code
+
+
+# The following functions will abstract the API calls:
+
+# 1. API calls for the cards
+
+def Cards_list(token, **args):
+    '''
+    Abstract API call card:list
+
+    Due to the API Limitation, it can only publish 500 records at once
+    A request has been made to Leankit to solve it
+    A solution is to play with offset to generate a list of all the cards
+    on the board
+
+    "pageMeta": {
+        "totalRecords": 582,
+        "offset": 500,
+        "limit": 500,
+        "startRow": 501,
+        "endRow": 582
+    }
+
+    '''
+    if set(('company', 'board')).issubset(args):
+        url = "https://{}{}".format(args['company'], web_card)
+        Bearer = "Bearer {}".format(token)
+        headers = {'Authorization': Bearer}
+        params = {'board': args['board'], 'limit': args['limit']}
+        response = get(url, headers=headers, params=params)
+
+        # First API Call
+        if response.status_code == 200:
+            try:
+                ret_lst = response.json()['cards']
+                total = response.json()['pageMeta']['totalRecords']
+                end_row = response.json()['pageMeta']['endRow']
+            except (json.decoder.JSONDecodeError, KeyError):
+                return {}
+
+            # Making sure all the cards are collected
+            while total != end_row:
+                params['offset'] = end_row
+                response = get(url, headers=headers, params=params)
+                try:
+                    ret_lst.extend(response.json()['cards'])
+                    total = response.json()['pageMeta']['totalRecords']
+                    end_row = response.json()['pageMeta']['endRow']
+                except (json.decoder.JSONDecodeError, KeyError):
+                    return {}
+
+        return ret_lst
+
+
+def Card_info(token, **args):
+    '''
+    Abstract Card:self
+    '''
+    if set(('company', 'card')).issubset(args):
+        url = "https://{}{}/{}".format(args['company'], web_card,
+                                       args['card'])
+        Bearer = "Bearer {}".format(token)
+        headers = {'Authorization': Bearer}
+        response = get(url, headers=headers)
+        if response.status_code == 200:
+            try:
+                return response.json()
+            except json.decoder.JSONDecodeError:
+                return {}
+
+
+def Cards_info(token, **args):
```

Comment:
seeing token get passed around makes me think you might want to put all this in a class, then it would be self.token...

though if that's the only shared data, maybe point.